web/tests/stream_data.test.cjs

```diff
@@ -1459,3 +1468,72 @@ test("can_access_stream_email", ({override}) => {
     page_params.is_spectator = true;
     assert.equal(stream_data.can_access_stream_email(social), false);
 });
+
+test("has_metadata_access", ({override}) => {
+    const social = {
+        subscribed: true,
+        color: "red",
+        name: "social",
+        stream_id: 2,
+        is_muted: false,
+        invite_only: true,
+        history_public_to_subscribers: false,
+        can_add_subscribers_group: nobody_group.id,
+        can_administer_channel_group: nobody_group.id,
+    };
+
+    assert.equal(stream_data.has_metadata_access(social), true);
+
+    social.is_web_public = true;
+    assert.equal(stream_data.has_metadata_access(social), true);
+    page_params.is_spectator = true;
+    assert.equal(stream_data.has_metadata_access(social), true);
+
+    social.is_web_public = false;
+    page_params.is_spectator = true;
+    assert.equal(stream_data.has_metadata_access(social), false);
+    page_params.is_spectator = false;
+
+    override(current_user, "is_admin", false);
+    assert.equal(stream_data.has_metadata_access(social), true);
+
+    // A user without the permission to administer a channel should not
+    // have metadata access to a private channel when unsubscribed.
+    social.subscribed = false;
+    assert.equal(stream_data.has_metadata_access(social), false);
+
+    // Realm admins should have metadata access to a private channel
+    // when unsubscribed.
+    override(current_user, "is_admin", true);
+    assert.equal(stream_data.has_metadata_access(social), true);
+    override(current_user, "is_admin", false);
+
+    // Channel admins should have metadata access to a private channel
+    // when unsubscribed.
+    social.can_administer_channel_group = me_group.id;
+    assert.equal(stream_data.has_metadata_access(social), true);
```

Comment:
The best way to write this sort of test is to have it check the value if `false` at the start of each stanza/commented block; that way, you are confirming for sure that you've not got test data pollution from previous changes not being reset properly making it return `true` for any reason other than whatever you just changed.

Reply:
Would it be ok to leave this as a followup? Otherwise I can tackle tomorrow